src/lib.rs

```diff
@@ -605,6 +715,11 @@ impl Context {
         }
     }
 
+    // TODO: Should we really keep `num_bytes` as `u16`?
```

Comment:
Maybe we should put `num_bytes` as `u16`, for this level of abstraction, as it is how the API defines it.
We can still define a higher-level function that takes an `usize` if needed.

Reply:
meaning that the `TODO` here is justified, not that you have to do the changes in this PR 😄